modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -1417,10 +1423,18 @@ def rebalance_partitions(cls, partitions):
                         ],
                         0,
                     )
+                    for obj in partitions[stop + 1]:
+                        obj._length_cache = partition_size - (
```

Comment:
Does [try_recompute_cache](https://github.com/modin-project/modin/blob/c67324b04b01e7d1f27a7c91565c488537891de1/modin/core/dataframe/pandas/partitioning/partition.py#L205) in the `mask` not cover us?

Reply:
Good question!

This should help in the case of a regular partition, but not in the case of a virtual one, where a materialization is performed before the mask is executed, which overwrites the length/width with remote references. However, it is possible to preserve caches when performing `force_materialization`, I suppose. What do you think about?

Reply:
Ah, I see. Let's leave the task of preserving caches through virtual partition `mask` for another PR.

Reply:
The issue for that: https://github.com/modin-project/modin/issues/4903